sdk/tests/deqp/framework/referencerenderer/rrRenderer.js

```diff
@@ -615,41 +950,34 @@ rrRenderer.drawTriangles = function(state, renderTarget, program, vertexAttribs,
         var v1 = rrRenderer.transformGLToWindowCoords(state, vertices[1]);
         var v2 = rrRenderer.transformGLToWindowCoords(state, vertices[2]);
 
+        rasterizer.init(v0, v1, v2);
+
+        // Culling
+        if ((state.cullMode == rrRenderState.CullMode.FRONT && rasterizer.m_face == rrDefs.FaceType.FACETYPE_FRONT) ||
+            (state.cullMode == rrRenderState.CullMode.BACK && rasterizer.m_face == rrDefs.FaceType.FACETYPE_BACK))
+        return;
+
         // Compute a conservative integer bounding box for the triangle
         var minX = Math.floor(Math.min(v0[0], v1[0], v2[0]));
         var maxX = Math.ceil(Math.max(v0[0], v1[0], v2[0]));
         var minY = Math.floor(Math.min(v0[1], v1[1], v2[1]));
         var maxY = Math.ceil(Math.max(v0[1], v1[1], v2[1]));
 
-        var fragmentPackets = [];
-        for (var i = minX; i <= maxX; i++) {
-            for (var j = minY; j <= maxY; j++) {
-                var x = i + 0.5;
-                var y = j + 0.5;
-
-                // A pixel is inside the triangle iff all its barycentric coordinates are non-negative
-                var b = rrRenderer.getBarycentricCoefficients([x, y], v0, v1, v2);
-                if (b[0] < 0 || b[1] < 0 || b[2] < 0) {
-                    continue;
-                }
-
-                var depth = v0[2] * b[0] + v1[2] * b[1] + v2[2] * b[2];
-                depth = depth * depthScale + depthBias;
-                if (!rrRenderer.clipTest(i, j, depth, state.viewport.rect))
-                    continue;
-                fragmentPackets.push(new rrFragmentOperations.Fragment(b, [i, j], depth));
-            }
-        }
```

Comment:
If a pixel is on the side of a triangle, one of its barycentric coordinates will be zero and will pass this check. 
When we use two triangles to draw a rectangle and let blending enabled, the pixels on the shared side of these two triangles will be blended twice and get wrong color.
e.g. Draw a rectangle at v0(0, 0),v1 (0, 128), v2(128, 128), v3(128, 0)
The formula of the diagnoal of the rectangle is x + y = 128 (v1 and v3)
i = 127, j = 0
x = 127.5, y=0.5; (x,y) is on the diagnoal, and (127,0) will be both in the fragment packets of two triangles
So does all pixels on the diagnoal; in SharedColorbufferTest of FboRenderTest, these points will be blended twice and a diagnoal will appear on the result picture.